src/snowflake/connector/log_configuration.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+from __future__ import annotations
+
+import inspect
+import json
+import logging
+import os
+
+from snowflake.connector import SnowflakeConnection
+from snowflake.connector.secret_detector import SecretDetector
+
+
+def is_full_path(path: str) -> bool:
+    return os.path.isabs(path)
+
+
+class EasyLoggingConfigPython:
+    def __init__(self, connection_parameters=None):
+        self.log_path = None
+        self.log_level = None
+        self.connection_parameters = connection_parameters
+        self.CLIENT_CONFIG_FILE = (
+            connection_parameters["CLIENT_CONFIG_FILE"]
+            if connection_parameters and "CLIENT_CONFIG_FILE" in connection_parameters
+            else None
+        )
+        self.SF_CLIENT_CONFIG_FILE = (
+            os.environ["SF_CLIENT_CONFIG_FILE"]
+            if "SF_CLIENT_CONFIG_FILE" in os.environ
+            else None
+        )
+        self.config_file_name = "sf_client_config.json"
+        self.log_file_name = "python_connector.log"
+        self.log_levels = logging._nameToLevel
```

Comment:
It's not great to use a non-public attribute of the `logging` module because that API isn't guaranteed to be stable. 

`logging.setLevel()` will throw a `ValueError` if given a bogus level string, so can we just rely on that to validate the level?  The other approach would be to do something like `getattr(logging, level)` on the string value and catch the resulting `AttributeError` when the level string is invalid.